src/deploy/osp_deployer/deployer.py

```diff
@@ -262,6 +276,45 @@ def deploy():
     logger.info("log : /auto_results/ ")
     sys.exit(ret_code)
 
+def validate_performance_optimization_params(
+        mariadb_max_connections,
+        innodb_buffer_pool_size,
+        innodb_buffer_pool_instances):
+    try:
+        mariadb_max_connections_int = int(mariadb_max_connections)
+    except Exception as error:
+        raise Exception("Please enter only integers for "
+                       "mariadb_max_connections in .ini file.")
+
+    if mariadb_max_connections_int < 1 or mariadb_max_connections_int > 100000:
+        logger.error("The value should be a number in range 1 to 100000")
+        raise Exception("Invalid param mariadb max connections")
+
+    if innodb_buffer_pool_size != 'dynamic':
+        pattern = '^\d+G$'
+        valid = re.match(pattern, innodb_buffer_pool_size)
+        if not valid:
+            logger.error(
+                "The given value of innodb_buffer_pool_size"
+                " in .ini file is invalid")
+            raise Exception("Invalid param innodb_buffer_pool_size")
+        innodb_buffer_pool_size = int(innodb_buffer_pool_size.replace("G",""))
+        if innodb_buffer_pool_size < 1:
+            logger.error(
+                "innodb_buffer_pool_instances not in range 20-100G")
+            raise Exception("Invalid param innodb_buffer_pool_size")
+    try:
+        innodb_buffer_pool_instances_int = int(innodb_buffer_pool_instances)
+    except Exception as error:
+        raise Exception(
+            "Please enter only integers for"
+            " innodb_buffer_pool_instances in .ini file.")
+    if innodb_buffer_pool_instances_int < 1 or \
+       innodb_buffer_pool_instances_int > 48:
+        logger.error(
+            "innodb_buffer_pool_instances not in range 1-48")
+        raise Exception("Invalid param innodb_buffer_pool_instances")
+
```

Comment:
I think we need to move those into the profiles where it would be best handled ,  create a is_int method  and an is in range one so those can be set in the .json file.

Reply:
Methods added to profile.py
JSON updated